packages/nextjs/components/ExampleUi/ContractInteraction/index.tsx

```diff
@@ -0,0 +1,19 @@
+import { CopyIcon } from "./CopyIcon";
+import { DiamondIcon } from "./DiamondIcon";
+import { HareIcon } from "./HareIcon";
+import { PurposeSetter } from "./PurposeSetter";
+import { UiDescription } from "./UiDescription";
+
+export const ContractInteraction = () => {
+  return (
+    <div className="flex bg-base-300 relative">
+      <DiamondIcon className="absolute top-24" />
+      <CopyIcon className="absolute bottom-0 left-36" />
+      <HareIcon className="absolute right-0 bottom-24" />
+      <div className="flex flex-col w-full mx-20">
+        <UiDescription />
+        <PurposeSetter />
```

Comment:
Personally, I think I would have inline both of these here. I guess I tend to split components a bit less. What is your mental model for it?

Don't change it! Just wanted to say it here so we can discuss and align better.

Reply:
It's easier to locate code/errors in smaller components. But to be honest, I thought it would be more code in that components :)